src/App.jsx

```diff
@@ -6,19 +6,25 @@ import AddTodoForm from './components/AddTodoForm';
 
 const Heading = () => <h1>Todo list</h1>
 
-const useSemiPersistentState = (key, initialValue) => {
-  const [value, setValue] = useState(
-    JSON.parse(localStorage.getItem(key)) || initialValue
-  );
+const App = () => {
+  const [todoList, setTodoList] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  useEffect(() => {
+    new Promise((resolve, reject) => 
+      setTimeout(
+        () => resolve({data: { todoList: JSON.parse(localStorage.getItem("savedTodoList")) } }),
+      2000
+      )
+    ).then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    })
```

Comment:
Right now it may not matter, but just in case, please add a `catch` clause, if anything just to add a console log as the assignment says

`.catch(error => console.log(‘error’, {error}))`

Reply:
added the catch at the end